pandas/plotting/_matplotlib/core.py

```diff
@@ -378,6 +423,12 @@ def _compute_plot_data(self):
                 label = "None"
             data = data.to_frame(name=label)
 
+        # GH15079 reconstruct data if by is defined
+        if self.by is not None:
+            self.subplots = True
+            grouped_data = data.groupby(self.by)
```

Comment:
Can you move the group inside `_reformat_grouped_data`. I think we want to have a function that does the whole transformation, not to split that part.

Reply:
okay, i thought it would be clearer to have the group outside. But I moved them in and add docs and example.